common/core/desktop/src/kmx/kmx_processor.cpp

```diff
@@ -353,20 +356,22 @@ int KMX_Processor::PostString(PKMX_WCHAR str, LPKEYBOARD lpkb, PKMX_WCHAR endstr
         m_actions.QueueAction(QIT_BELL, 0);
         break;
       case CODE_CONTEXT:        // copy the context to the output
-        for(q = m_miniContext; *q; q++) {
-          m_actions.QueueAction(QIT_CHAR, *q);
-        }
+      {
+        KMX_BOOL ignoreOutputKeystroke;
+        PostString(m_miniContext, lpkb, (PKMX_WCHAR) u16chr(m_miniContext, 0), &ignoreOutputKeystroke);
         break;
+      }
       case CODE_CONTEXTEX:
+      {
+        KMX_BOOL ignoreOutputKeystroke;
         p++;
-        for(q = m_miniContext, i = 0; *q && i < *p-1; i++, q=incxstr(q));
-        if(*q) {
-          m_actions.QueueAction(QIT_CHAR, *q);
-          if(Uni_IsSurrogate1(*q) && Uni_IsSurrogate2(*(q+1))) {
-            m_actions.QueueAction(QIT_CHAR, *(q+1));
-          }
+        for (q = m_miniContext, i = m_miniContextIfLen; *q && i < *p - 1; i++, q = incxstr(q));
+        if (*q) {
+          temp = incxstr(q);
+          PostString(q, lpkb, temp, &ignoreOutputKeystroke);
```

Comment:
This change ensures we cope with deadkeys which can also be found in the `m_miniContext` buffer. It outputs only the first element found at `q`.